contrib/hashicorp/consul/consul_test.go

```diff
@@ -0,0 +1,104 @@
+package consul
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/mocktracer"
+
+	consul "github.com/hashicorp/consul/api"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestMain(m *testing.M) {
+	_, ok := os.LookupEnv("INTEGRATION")
+	if !ok {
+		fmt.Println("--- SKIP: to enable integration test, set the INTEGRATION environment variable")
+		os.Exit(0)
+	}
+	os.Exit(m.Run())
+}
+
+func TestClient(t *testing.T) {
+	assert := assert.New(t)
+	mt := mocktracer.Start()
+	defer mt.Stop()
+
+	client, err := NewClient(consul.DefaultConfig())
+	if err != nil {
+		panic(err)
+	}
+
+	assert.NotNil(client)
+	assert.Nil(err)
+}
+
+func TestClientError(t *testing.T) {
+	assert := assert.New(t)
+	mt := mocktracer.Start()
+	defer mt.Stop()
+
+	var config = &consul.Config{
+		Address: "badscheme://baduri",
+	}
+
+	client, err := NewClient(config)
+	assert.Nil(client)
+	assert.Error(err)
+
+}
+
+func TestClientKV(t *testing.T) {
+	assert := assert.New(t)
+	mt := mocktracer.Start()
+	defer mt.Stop()
+
+	client, _ := NewClient(consul.DefaultConfig())
+	kv := client.KV()
+
+	assert.Equal(client.ctx, kv.ctx)
+}
+
+func TestKV(t *testing.T) {
+	key := "test.key"
+	pair := &consul.KVPair{Key: key, Value: []byte("test_value")}
+	testCases := map[string]func(kv *KV){
```

Comment:
Same here, can we inline this map? I feel it's easier to follow. We don't need the variable -- it's not used anywhere else.

Reply:
Done for tests & benchmarks